[jessicawyz] iP

jessicawyz opened the pull request:
**Echo**

> “Your mind is for having ideas, not holding them.” – David Allen [[source](https://dansilvestre.com/productivity-quotes](url)

Echo is your best friend and secretary, helping you track your schedule and tasks. It's,


- text-based

- easy to learn

- SUPER FAST AND EASY to use

**Start using it NOW,**

1. download it (https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features:

 Managing tasks:

- [ ] 1.  Deadlines
- [ ] 2. Events
- [ ] 3. Todos
- [ ] 4.  Reminders (coming soon)

use it to practice your JAVA:
```
    public static void main(String[] args) {
        Echo echo = new Echo();
        Ui.greetUser();
        echo.start();
        Ui.endConversation();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
"Level-0 commit"

Commit:
"level 1 commit"

Commit:
"REAL level 1"

Commit:
"level 2"

Commit:
"level 3, added task and task list classes"

Commit:
"added 3 clases extending tasks, level 4"

Commit:
added the testing parts

Commit:
Level 5 error handling

Commit:
delete tasks

Commit:
level 7 added file that stores tasklist

Commit:
level 7 added file that stores tasklist

Commit:
Merge commit 'a6ab2d8d185bce5e8cfc53e1438ee86af6ebc24c'

* commit 'a6ab2d8d185bce5e8cfc53e1438ee86af6ebc24c':
  level 7 added file that stores tasklist

Commit:
Level 8 data time + bug fix for level 7 changed to save all tasks when exit, reload when open bot again

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level 8 data time + bug fix for level 7 changed to save all tasks when exit, reload when open bot again

Commit:
oop changes, taskmanager is the same as tasklist

Commit:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  oop changes, taskmanager is the same as tasklist

Commit:
packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Add junit testes and gradle support.

Commit:
Add junit testes and gradle support.

Commit:
Add build.gradle information

Commit:
Add Javadoc comments

The Parser, Storage and Deadline files have class and method comments

Commit:
Add Javadoc comments

Add Javadoc comments and headers to the rest of the files

Commit:
Add indentation according to standards

Commit:
Add indentation to test files

Add indentation according to coding standard

Commit:
Add level 9 increment

Find tasks is available by finding in the arraylist

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add level 9 increment

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add Javadoc comments
  Add Javadoc comments

# Conflicts:
#	src/main/java/Echo/TaskManager.java

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add indentation to test files
  Add indentation according to standards

# Conflicts:
#	src/main/java/Echo/TaskManager.java

Commit:
Add checkstyle files

Add files including checkstyle.xml and suppressions.xml

ThatGirlJam commented on src/main/java/Echo/Storage/Storage.java:
```diff
@@ -0,0 +1,144 @@
+import Echo.Task.Deadline;
+import Echo.Task.Event;
+
+import java.io.*;
```
Perhaps consider not using a wildcard import?

ThatGirlJam commented on src/main/java/Echo/Task/Task.java:
```diff
@@ -0,0 +1,82 @@
+/*
```
I like the use of JavaDocs and Descriptions!

ThatGirlJam commented on src/main/java/Echo/TaskManager.java:
```diff
@@ -0,0 +1,333 @@
+    public void executeCommand(String command) {
+        String[] tokens = command.split(" ", 2);
+
+        switch (tokens[0].toLowerCase()) {
```
Perhaps tweak this such that "switch" and "case" are the same number of indents, according to the coding standard? This can be done using the IDE's settings 👍

ThatGirlJam commented on src/main/java/Echo/TaskManager.java:
```diff
@@ -0,0 +1,333 @@
+            }
+            String taskDescription = taskTokens[1].trim();
+
+            switch (taskType) {
```
Likewise, may need to change this to align "switch" and case"?

Emberlynn-Loo replied:
I do like that for your switch cases, you specified each case instead of "case 1,2,3 etc" as most people do, makes it more readable.

ThatGirlJam reviewed (commented):
Some minor cosmetic changes to consider, but LGTM! 👍

Commit:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add checkstyle files

Commit:
Add basic GUI

Add changes to echo main class and ui

Commit:
Add bug fix marking tasks

Mark done and undone

Emberlynn-Loo commented on src/main/java/Echo/Echo.java:
```diff
@@ -0,0 +1,65 @@
+ *
+ */
+
+package Echo;
+
+import Echo.Storage.Storage;
+import Echo.Ui.Ui;
```
Consider using lowercase for package names (echo) as per Java naming conventions.

```suggestion
package echo;

import echo.Storage.Storage;
import echo.Ui.Ui;
```

Emberlynn-Loo commented on src/main/java/Echo/Echo.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    public static void main(String[] args) {
+        Echo echo = new Echo();
+        Ui.greetUser();
+        echo.start();
+        Ui.endConversation();
```
Consider using instance methods for Ui class instead of calling static methods directly?

Emberlynn-Loo commented on src/main/java/Echo/Storage/Storage.java:
```diff
@@ -0,0 +1,144 @@
+                default:
+                    throw new IllegalArgumentException("Invalid task type in file!");
+            }
+        } catch (IllegalArgumentException e) {
```
I like how you handled the exceptions!

Emberlynn-Loo reviewed (commented):
Overall, the code is well-structured and easy to follow, with clear separation of concerns between classes. Some nitpicking would be the importing of classes and capitalisation of your packages and imports. Coding has good readability tho, great work!!

Commit:
Add GUI updates

Add changes and improvements to GUI. Improve the UI and images.
Fix bugs.

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI updates
  Add bug fix marking tasks
  Add basic GUI

Commit:
Add changes to build.gradle

Add new shadowjar component to build.gradle.
This is to create a new jar file called duke2.

Commit:
Add changes to build.gradle

Add new shadowjar component to build.gradle. This is to create a new jar file called duke2.

Commit:
Add assertions

Add assertions to multiple files to check assumptions

Commit:
Add assertions

Add more assertions in taskManager class.

Check assumptions that important variables are not null.

Commit:
Improve code quality

Improve arrowhead nesting with extraction method.

Commit:
Add new code quality changes

Add changes according to java standards.

Commit:
Add streams usages

Add streams uses in echo and taskManager methods.

jessicawyz committed:
Merge branch 'master' into branch-A-Streams

jessicawyz committed:
Merge pull request #2 from jessicawyz/branch-A-Assertions

Add assertions

jessicawyz committed:
Merge branch 'master' into branch-A-Streams

jessicawyz committed:
Merge pull request #4 from jessicawyz/branch-A-Streams

Add streams usages

jessicawyz committed:
Merge branch 'master' into branch-A-CodeQuality

jessicawyz committed:
Merge pull request #3 from jessicawyz/branch-A-CodeQuality

Branch a code quality

Commit:
Merge branch 'master' of https://github.com/jessicawyz/ip

* 'master' of https://github.com/jessicawyz/ip:
  Add new code quality changes
  Improve code quality

Commit:
Merge branch 'branch-A-CodeQuality' of https://github.com/jessicawyz/ip into branch-A-CodeQuality

* 'branch-A-CodeQuality' of https://github.com/jessicawyz/ip:
  Add streams usages
  Add assertions
  Add changes to build.gradle

Commit:
Merge branch 'branch-A-Streams' of https://github.com/jessicawyz/ip into branch-A-Streams

* 'branch-A-Streams' of https://github.com/jessicawyz/ip:
  Add assertions
  Add changes to build.gradle

Commit:
Add extension

Add scheduling conflicts and bot responses.

Commit:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Add extension

Commit:
Merge branch 'branch-A-Streams'

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Change to adhere to java coding standards

Improve GUI, add java doc, change formatting.

Commit:
Add readme

Commit:
Add bug fixes

Commit:
Fix javafx

Commit:
Fix bug
